### PR TITLE
minor adjustments in the sample apiserver around resource creation.

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: This PR improves `NewREST` method to support an error and nor-error path. The method is then wrapped in a convenient function which panics on error.

**Release note**:

```
NONE
```

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
@deads2k hopefully the changes will make you happy :)

File: staging/src/k8s.io/sample-apiserver/pkg/apiserver/apiserver.go [hunk 1]
```diff
@@ -103,8 +104,8 @@ func (c completedConfig) New() (*WardleServer, error) {
 	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(wardle.GroupName, registry, Scheme, metav1.ParameterCodec, Codecs)
 	apiGroupInfo.GroupMeta.GroupVersion = v1alpha1.SchemeGroupVersion
 	v1alpha1storage := map[string]rest.Storage{}
-	v1alpha1storage["flunders"] = flunderstorage.RESTInPeace(Scheme, c.GenericConfig.RESTOptionsGetter)
-	v1alpha1storage["fischers"] = fischerstorage.RESTInPeace(Scheme, c.GenericConfig.RESTOptionsGetter)
+	v1alpha1storage["flunders"] = wardleregistry.RESTInPeace(flunderstorage.NewREST(Scheme, c.GenericConfig.RESTOptionsGetter))
+	v1alpha1storage["fischers"] = wardleregistry.RESTInPeace(fischerstorage.NewREST(Scheme, c.GenericConfig.RESTOptionsGetter))
```
Comment (liggitt) [inline staging/src/k8s.io/sample-apiserver/pkg/apiserver/apiserver.go]: at first glace, this is moving the constructor to a package I didn't expect

Comment (p0lyn0mial) [inline staging/src/k8s.io/sample-apiserver/pkg/apiserver/apiserver.go]: if it helps we could change the pkg name to `wardlestorage`. Notice that `RESTInPeace` is just a wrapper. To give you more context https://github.com/kubernetes/kubernetes/pull/48444#issuecomment-313105734

Comment (deads2k) [inline staging/src/k8s.io/sample-apiserver/pkg/apiserver/apiserver.go]: > at first glace, this is moving the constructor to a package I didn't expect

It's a wrapper we're trying out here to allow errors from REST construction  without panics.  We did a similar thing at a lower level in openshift and I wasn't a huge fan of where we put it.

**Comment by p0lyn0mial** [comment]
I think that the tests were very unstable at that time. I will rebase against the master.

**Comment by p0lyn0mial** [comment]
@deads2k the bot is not obedient :( please, retest and retag.

**Comment by deads2k** [comment]
@eparis how do we get @p0lyn0mial added to the whitelist to be tested?

**Pull request merged**